assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -63,6 +64,41 @@ interactivityApiStore( {
 	state: {},
 	effects: {
 		woocommerce: {
+			watchForChangesOnAddToCartForm: ( store: Store ) => {
+				const variableProductCartForm = document.querySelector(
+					`form[data-product_id="${ store.context.woocommerce.productId }"]`
+				);
+
+				if ( ! variableProductCartForm ) {
+					return;
+				}
+
+				const observer = new MutationObserver( function ( mutations ) {
+					for ( const mutation of mutations ) {
+						const mutationTarget = mutation.target as HTMLElement;
+						const currentImageAttribute =
+							mutationTarget.getAttribute( 'current-image' );
+						if (
+							mutation.type === 'attributes' &&
+							currentImageAttribute &&
+							store.context.woocommerce.visibleImagesIds.includes(
+								currentImageAttribute
+							)
+						) {
+							store.context.woocommerce.selectedImage =
+								currentImageAttribute;
+						}
+					}
+				} );
+
+				observer.observe( variableProductCartForm, {
+					attributes: true,
+				} );
+
+				return () => {
+					observer.disconnect();
```

Comment:
Out of curiosity, I understand this is a cleanup function, but are we ever actually disconnecting the observer?